hn2016_falwa/xarrayinterface.py

```diff
@@ -0,0 +1,534 @@
+        """Collect the output of `compute_lwa_and_barotropic_fluxes` in a dataset.
+
+        See :py:meth:`.oopinterface.QGField.compute_lwa_and_barotropic_fluxes`.
+
+        Returns
+        -------
+        xarray.Dataset
+        """
+        # Call compute_lwa_and_barotropic_fluxes on all QGField objects
+        out_fields = _map_collect(
+            lambda field: field.compute_lwa_and_barotropic_fluxes(northern_hemisphere_results_only),
+            self._fields,
+            ["adv_flux_f1", "adv_flux_f2", "adv_flux_f3", "convergence_zonal_advective_flux",
+                "divergence_eddy_momentum_flux", "meridional_heat_flux", "lwa_baro", "u_baro",
+                "lwa"],
+            postprocess=np.asarray
+        )
+        # Take the first field to extract coordinates and metadata
+        _field = self.fields[0]
+        # Prepare coordinate-related data for the output
+        if northern_hemisphere_results_only:
+            _ylat = _field.ylat[(_field.equator_idx - 1):]            
+        else:
+            _ylat = _field.ylat
+        # 2D data, function of latitude and longitude
+        out_dims_2d = (*self._other_dims, "ylat", "xlon")
+        out_shape_2d = (*self._other_shape, _ylat.size, _field.xlon.size)
+        # 3D data, function of height, latitude and longitude
+        out_dims_3d = (*self._other_dims, "height", "ylat", "xlon")
+        out_shape_3d = (*self._other_shape, _field.height.size, _ylat.size, _field.xlon.size)
+        # Combine all outputs into a dataset, reshape to restore the original
+        # other dimensions that were flattened earlier
+        return xr.Dataset(
+            data_vars={
+                "adv_flux_f1": (out_dims_2d, out_fields["adv_flux_f1"].reshape(out_shape_2d)),
+                "adv_flux_f2": (out_dims_2d, out_fields["adv_flux_f2"].reshape(out_shape_2d)),
+                "adv_flux_f3": (out_dims_2d, out_fields["adv_flux_f3"].reshape(out_shape_2d)),
+                "convergence_zonal_advective_flux": (out_dims_2d, out_fields["convergence_zonal_advective_flux"].reshape(out_shape_2d)),
+                "divergence_eddy_momentum_flux": (out_dims_2d, out_fields["divergence_eddy_momentum_flux"].reshape(out_shape_2d)),
+                "meridional_heat_flux": (out_dims_2d, out_fields["meridional_heat_flux"].reshape(out_shape_2d)),
+                "lwa_baro": (out_dims_2d, out_fields["lwa_baro"].reshape(out_shape_2d)),
+                "u_baro": (out_dims_2d, out_fields["u_baro"].reshape(out_shape_2d)),
+                "lwa": (out_dims_3d, out_fields["lwa"].reshape(out_shape_3d)),
+            },
+            coords={
+                **self._other_coords,
+                "height": _field.height,
+                "ylat": _ylat,
+                "xlon": _field.xlon,
+            },
+            attrs=self.attrs
+        )
+
+    # The new routines so far only seem to work for 1°-resolution data and the northern hemisphere
```

Comment:
Yes. It's true that the additional direct inversion algorithm assumes 1°-resolution data (since we were in a hurry submitting the manuscript and archiving the code is a requirement). Let me add an issue ticket for future work generalizing this to arbitrary resolution.